model/testresult/testresult.go

```diff
@@ -0,0 +1,81 @@
+package testresult
+
+import (
+	"github.com/evergreen-ci/evergreen/db"
+	"github.com/evergreen-ci/evergreen/db/bsonutil"
+	mgo "gopkg.in/mgo.v2"
+	"gopkg.in/mgo.v2/bson"
+)
+
+const (
+	collection = "testresults"
+)
+
+// TestResult contains test data for a task.
+type TestResult struct {
+	ID        bson.ObjectId `bson:"_id" json:"_id"`
+	Status    string        `json:"status" bson:"status"`
+	TestFile  string        `json:"test_file" bson:"test_file"`
+	URL       string        `json:"url" bson:"url,omitempty"`
+	URLRaw    string        `json:"url_raw" bson:"url_raw,omitempty"`
+	LogID     string        `json:"log_id,omitempty" bson:"log_id,omitempty"`
+	LineNum   int           `json:"line_num,omitempty" bson:"line_num,omitempty"`
+	ExitCode  int           `json:"exit_code" bson:"exit_code"`
+	StartTime float64       `json:"start" bson:"start"`
+	EndTime   float64       `json:"end" bson:"end"`
+
+	// Together, TaskID and Execution identify the task which created this TestResult
+	TaskID    string `bson:"task_id" json:"task_id"`
+	Execution int    `bson:"task_execution" json:"task_execution"`
+
+	// LogRaw is not persisted to the database
+	LogRaw string `json:"log_raw" bson:"log_raw,omitempty"`
+}
+
+var (
+	// BSON fields for the task struct
+	testResultStatusKey    = bsonutil.MustHaveTag(TestResult{}, "Status")
```

Comment:
run linter on these

Reply:
I'm exporting these in EVG-2053, since they will be accessed outside the package.